### PR TITLE
SW-2517 Show completed tests as 'marked as complete'

### PR DESCRIPTION
**Pull request by karthikbtf**
- the checkbox was not being initialized for already saved tests
- was mostly working for newly created/edited tests

Terraware App 2022-12-12 15-00-27

### CODE REVIEW

**Comment by charlie-tf** [comment]
Maybe out of the scope of this PR, but there appear to be some issues with the state of the checkbox resetting itself when an observation is cancelled,
Kapture 2022-12-12 at 21 36 15

It might also not be resetting when closing the dialog. I was having some cases where ending a test, then creating a new test caused the new test to end immediately, but I couldn't quite get something to break there consistently.

**Review by constanzauanini — APPROVED** [review]

**Comment by karthikbtf** [comment]
> Maybe out of the scope of this PR, but there appear to be some issues with the state of the checkbox resetting itself when an observation is cancelled, Kapture 2022-12-12 at 21 36 15 [ Kapture 2022-12-12 at 21 36 15 ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif) [ ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> 
> It might also not be resetting when closing the dialog. I was having some cases where ending a test, then creating a new test caused the new test to end immediately, but I couldn't quite get something to break there consistently. EDIT: it seems to reproduce if I attempt to create a new test immediately after deleting a completed test. Kapture 2022-12-12 at 21 51 24 [ Kapture 2022-12-12 at 21 51 24 ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif) [ ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)

Thanks for pointing this out, I will take a look, clearly something is broken.

**Comment by karthikbtf** [comment]
> > Maybe out of the scope of this PR, but there appear to be some issues with the state of the checkbox resetting itself when an observation is cancelled, Kapture 2022-12-12 at 21 36 15
> >       
> >         [
> >           
> >             Kapture 2022-12-12 at 21 36 15
> >           
> >         ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> >         
> >        [ Kapture 2022-12-12 at 21 36 15 ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> >       
> >         [
> >           
> >             Kapture 2022-12-12 at 21 36 15
> >           
> >         ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> >         
> >        [ ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> > It might also not be resetting when closing the dialog. I was having some cases where ending a test, then creating a new test caused the new test to end immediately, but I couldn't quite get something to break there consistently. EDIT: it seems to reproduce if I attempt to create a new test immediately after deleting a completed test. Kapture 2022-12-12 at 21 51 24
> >       
> >         [
> >           
> >             Kapture 2022-12-12 at 21 51 24
> >           
> >         ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> >         
> >        [ Kapture 2022-12-12 at 21 51 24 ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> >       
> >         [
> >           
> >             Kapture 2022-12-12 at 21 51 24
> >           
> >         ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> >         
> >         
> >           
> >             
> >               
> >             
> >             
> >               
> >               
> >             
> >           
> >           [
> >             
> >               
> >             
> >           ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> >         
> >        [ ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> 
> Thanks for pointing this out, I will take a look, clearly something is broken.

This is now fixed, there was a bug where if you marked a test as completed and you deleted the test row,,, the 'test completed' status stuck during save. And subsequent edits were disabled.

**Comment by karthikbtf** [comment]
> > > Maybe out of the scope of this PR, but there appear to be some issues with the state of the checkbox resetting itself when an observation is cancelled, Kapture 2022-12-12 at 21 36 15
> > >       
> > >         [
> > >           
> > >             Kapture 2022-12-12 at 21 36 15
> > >           
> > >         ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> > >         
> > >         
> > >           
> > >             
> > >               
> > >             
> > >             
> > >               
> > >               
> > >             
> > >           
> > >           [
> > >             
> > >               
> > >             
> > >           ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> > >         
> > >       
> > > ```
> > >     [
> > >       
> > >         Kapture 2022-12-12 at 21 36 15
> > >       
> > >     ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> > >     
> > >     
> > >       
> > >         
> > >           
> > >         
> > >         
> > >           
> > >           
> > >         
> > >       
> > >       [
> > >         
> > >           
> > >         
> > >       ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> > >     
> > >    [ Kapture 2022-12-12 at 21 36 15 ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> > >   
> > >     [
> > >       
> > >         Kapture 2022-12-12 at 21 36 15
> > >       
> > >     ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> > >     
> > >     
> > >       
> > >         
> > >           
> > >         
> > >         
> > >           
> > >           
> > >         
> > >       
> > >       [
> > >         
> > >           
> > >         
> > >       ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> > >     
> > >    [ ](https://user-images.githubusercontent.com/114949086/207227828-1082d336-6b82-4b75-98b6-f35ecc8204e4.gif)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > It might also not be resetting when closing the dialog. I was having some cases where ending a test, then creating a new test caused the new test to end immediately, but I couldn't quite get something to break there consistently. EDIT: it seems to reproduce if I attempt to create a new test immediately after deleting a completed test. Kapture 2022-12-12 at 21 51 24
> > >       
> > >         [
> > >           
> > >             Kapture 2022-12-12 at 21 51 24
> > >           
> > >         ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> > >         
> > >         
> > >           
> > >             
> > >               
> > >             
> > >             
> > >               
> > >               
> > >             
> > >           
> > >           [
> > >             
> > >               
> > >             
> > >           ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> > >         
> > >       
> > > ```
> > >     [
> > >       
> > >         Kapture 2022-12-12 at 21 51 24
> > >       
> > >     ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> > >     
> > >     
> > >       
> > >         
> > >           
> > >         
> > >         
> > >           
> > >           
> > >         
> > >       
> > >       [
> > >         
> > >           
> > >         
> > >       ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> > >     
> > >    [ Kapture 2022-12-12 at 21 51 24 ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> > >   
> > >     [
> > >       
> > >         Kapture 2022-12-12 at 21 51 24
> > >       
> > >     ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> > >     
> > >     
> > >       
> > >         
> > >           
> > >         
> > >         
> > >           
> > >           
> > >         
> > >       
> > >       [
> > >         
> > >           
> > >         
> > >       ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> > >     
> > >    [ ](https://user-images.githubusercontent.com/114949086/207229640-48a4abd1-5cb1-4e68-bb88-d006d67c0647.gif)
> > > ```
> > 
> > 
> > Thanks for pointing this out, I will take a look, clearly something is broken.
> 
> This is now fixed, there was a bug where if you marked a test as completed and you deleted the test row,,, the 'test completed' status stuck during save. And subsequent edits were disabled.

Purely client-side issue.

File: src/components/NurseryWithdrawals/ReassignmentRenderer.tsx [hunk 1]
```diff
@@ -78,6 +78,7 @@ export default function ReassignmentRenderer({ plots, setReassignment }: Reassig
           }}
           isEqual={(option: any, selected: any) => option?.value === selected?.value}
           label={''}
+          placeholder={strings.SELECT}
```
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/ReassignmentRenderer.tsx:81]: i snuck these in

File: src/components/accession2/viabilityTesting/NewViabilityTestModal.tsx [hunk 2]
```diff
@@ -302,6 +318,10 @@ export default function NewViabilityTestModal(props: NewViabilityTestModalProps)
       const updatedResults = [...record.testResults];
       updatedResults.splice(index, 1);
       onChange('testResults', updatedResults);
+      if (updatedResults.length === 0) {
```
Comment (karthikbtf) [inline src/components/accession2/viabilityTesting/NewViabilityTestModal.tsx:321]: this is the fix

Comment (constanzauanini) [reply src/components/accession2/viabilityTesting/NewViabilityTestModal.tsx:321]: Excellent!

**Review by constanzauanini — APPROVED** [review]

**Pull request merged**